File: Array/413.Arithmetic_Slices/Arithmetic_Slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        if(nums.size() < 3) return 0;
        
        int n = nums[1] - nums[0], cnt = 2, total = 0;
        
        for(int i = 2; i < nums.size(); i++)
        {
            if(nums[i] - nums[i - 1] == n) cnt++;
            
            else
            {
                if(cnt >= 3)
                {
                    cnt -= 2;
                    total += ((cnt * cnt + cnt) / 2);
                }
                
                n = nums[i] - nums[i - 1];
                cnt = 2;
            }
        }
        
        if(cnt > 2)
        {
            cnt -= 2;
            total += ((cnt * cnt + cnt) / 2);
        }
        
        return total;
    }
};

File: Array/413.Arithmetic_Slices/Arithmetic_Slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        
        if len(nums) <= 2: return 0
        
        diff = nums[1] - nums[0]; cnt = 2; ans = 0
        
        for i in range(2, len(nums)):
            
            if(nums[i] - nums[i - 1] == diff):
                cnt += 1
                
            else:
                
                if(cnt >= 3):
                    cnt -= 2
                    ans += (pow(cnt , 2) + cnt) // 2
                     
                diff = nums[i] - nums[i - 1]
                cnt = 2
                
        
        if cnt >= 3:
            cnt -= 2
            ans += (pow(cnt , 2) + cnt) // 2
            
        return ans
    